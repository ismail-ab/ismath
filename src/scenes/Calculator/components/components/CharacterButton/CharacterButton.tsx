import * as React from 'react';
import {
  Button,
  Grid,
  PropTypes
} from '@material-ui/core';
import { CalculatorContext } from '../../../contexts';

interface ICharacterButtonOwnProps {
  characterToDisplay: string;
  color?: PropTypes.Color;
}

const CharacterButton: React.SFC<ICharacterButtonOwnProps> = (props) => {
  const {
    characterToDisplay,
    color = 'primary'
  } = props;

  return (
    <CalculatorContext.Consumer>
      {({ concatenateCharactersOrCompute }) => (
        <Grid
          item={true}
          xs={4}
        >
          <Button
            color={color}
            variant="raised"
            onClick={() => concatenateCharactersOrCompute(characterToDisplay)}
            style={{
              height: '4em',
              borderRadius: '0',
              fontSize: '1.2em'
            }}
          >
            {characterToDisplay}
          </Button>
        </Grid>
      )}
    </CalculatorContext.Consumer>
  );
};

export default CharacterButton;