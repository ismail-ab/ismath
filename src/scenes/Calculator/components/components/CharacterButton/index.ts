export { default as CharacterButton } from './CharacterButton';