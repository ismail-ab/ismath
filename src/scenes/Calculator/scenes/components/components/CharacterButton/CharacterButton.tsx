import * as React from 'react';
import {
  Button,
  Grid,
  PropTypes
} from '@material-ui/core';
import { CalculatorContext } from '../../../contexts';

interface ICharacterButtonOwnProps {
  characterToDisplay: string;
  color?: PropTypes.Color;
}

const CharacterButton: React.SFC<ICharacterButtonOwnProps> = (props) => {
  const {
    characterToDisplay,
    color = 'primary'
  } = props;

  const handleClick = (concatenateCharacters: Function, _character: string) => {
    concatenateCharacters(_character);
  };

  return (
    <CalculatorContext.Consumer>
      {({ concatenateCharacters }) => (
        <Grid
          item={true}
          xs={4}
        >
          <Button
            color={color}
            variant="raised"
            onClick={() => handleClick(concatenateCharacters, characterToDisplay)}
            style={{
              height: '4em',
              borderRadius: '0',
              fontSize: '1.2em'
            }}
          >
            {characterToDisplay}
          </Button>
        </Grid>
      )}
    </CalculatorContext.Consumer>
  );
};

export default CharacterButton;