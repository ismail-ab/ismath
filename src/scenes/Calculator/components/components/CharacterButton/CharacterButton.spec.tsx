import { } from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import { PropTypes } from '@material-ui/core';
import CharacterButton from './CharacterButton';

describe('CharacterButton', () => {
  it('renders correctly', () => {
    const color: PropTypes.Color = 'primary';
    const minProps = {
      characterToDisplay: '=',
      color
    };
    const wrapper = shallow(<CharacterButton {...minProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
