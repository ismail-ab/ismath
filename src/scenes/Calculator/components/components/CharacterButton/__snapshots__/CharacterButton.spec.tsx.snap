// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterButton renders correctly 1`] = `<[object Object] />`;
